src/command/process_update_task.rs

```diff
@@ -110,18 +123,40 @@ impl UpdateProcessor {
         }
     }
 
-    pub async fn process(&self) -> Result<(), BotError> {
+    pub async fn process(&self) -> Result<Option<ScheduleWeatherTask>, BotError> {
```

Comment:
since your db_pool is global and you won't be creating new connection. maybe it's fine to enqueue a new task directly inside of this impl

btw it seems ScheduleWeatherTask is not used anymore?

Reply:
Deleted.